app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id', counter_cache: 'post_counter'
+  has_many :comments, foreign_key: 'post_id', dependent: :destroy
+  has_many :likes, foreign_key: 'post_id', dependent: :destroy
+
+  def most_recent_comment
+    comments.order('created_at DESC').limit(5)
+  end
+
+  def update_post_counter
+    user.update(post_counter: user.posts.size)
+  end
+end
```

Comment:
- [ ] [OPTIONAL] Great implementation @collinsmezie 👍🏽 However, I would like to suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to the database, then make the update_post_counter method private. You can try it out:

```ruby
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```